packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -196,10 +202,56 @@ export default class CallDiagnosticMetrics {
   }
 
   /**
-   * TODO: NOT IMPLEMENTED
+   * Return Client Event payload by client error code
+   * @param clientErrorCode
+   * @returns
+   */
+  public getErrorPayloadForClientErrorCode(clientErrorCode: number): ClientEventError {
+    let error: ClientEventError;
+
+    if (clientErrorCode) {
+      const partialParsedError = CLIENT_ERROR_CODE_TO_ERROR_PAYLOAD[clientErrorCode];
+
+      if (partialParsedError) {
+        error = merge(
+          {fatal: true, shownToUser: false, name: 'other', category: 'other'}, // default values
+          partialParsedError
+        );
+
+        return error;
+      }
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Generate error payload for Client Event
    * @param rawError
    */
-  generateErrorPayload(error: any) {
+  generateClientEventErrorPayload({
+    rawError,
+    parsedError,
+  }: {
+    rawError?: any;
+    parsedError?: ClientEventError;
+  }) {
+    // sometimes we compute the error on the go
+    if (parsedError) {
+      return parsedError;
+    }
```

Comment:
I think, this break the "A function should do one thing and do it well." rule.

The caller can check and use `parsedError` otherwise call `generateClientEventErrorPayload`

Reply:
yeah, i realise now it's useless